kiota_abstractions/api_error.py

```diff
@@ -1,4 +1,16 @@
 class APIError(Exception):
+    """The base class for all API errors."""
 
-    def __init__(self, message: str) -> None:
-        super().__init__(message)
+    def __init__(self, status_code: int, message: str) -> None:
+        self._response_status_code = status_code
+        self._message = message
```

Comment:
Do we have to explicitly set the message when calling the super constructor?

Reply:
Yes, in order to show our custom error message, otherwise the super class will only show a generic error message